tap_google_analytics/client.py

```diff
@@ -13,12 +13,29 @@
 
 LOGGER = singer.get_logger()
 
+def is_retryable_403(e):
+    response = e.response
+    if not _is_json(response):
+        return False
+
+    retryable_errors = ["userRateLimitExceeded", "rateLimitExceeded", "quotaExceeded"]
+    error_reasons = [error.get('reason') for error in response.json().get('errors',[])]
+    for retryable_error in retryable_errors:
+        if retryable_error in error_reasons:
+            return True
+    return False
+
 def should_giveup(e):
-    if e.response.status_code == 429:
```

Comment:
I'd like to propose that any time we write a function for `giveup` that we add docs that remind us what happens when you return true or false.

Reply:
Can do